### PR TITLE
Extract serde-varint crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`solana_program::serde_varint` is another module that makes it hard to split up solana-program

#### Summary of Changes
Give it its own crate and re-export with deprecation notice

### CODE REVIEW

**Comment by kevinheavey** [comment]
This circular dependency false positive in CI will go away when #1796 is merged

File: programs/sbf/Cargo.lock [hunk 1]
```diff
@@ -6211,6 +6214,13 @@ dependencies = [
  "solana-tpu-client",
 ]
 
+[[package]]
+name = "solana-serde-varint"
+version = "2.1.0"
+dependencies = [
+ "serde",
+]
```
Comment (joncinque) [inline programs/sbf/Cargo.lock:6364]: This is really nice

File: sdk/serde-varint/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "solana-serde-varint"
+description = "Integers that serialize to variable size"
```
Comment (joncinque) [inline sdk/serde-varint/Cargo.toml]: nit: maybe we should call out "solana" in some way:
```suggestion
description = "Solana Definitions for Integers that serialize to variable size"
```

**Review by joncinque — COMMENTED** [review]
Just the one nit about the crate description, then this should be good to go! solana-serde-varint is already owned by anza-team, so we're good there.

**Review by joncinque — APPROVED** [review]
Thanks for fixing this all up so quickly!

**Pull request merged**